src/components/Hits/withClickAnalytics.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'preact-compat';
+import PropTypes from 'prop-types';
+
+export default C =>
+  class WithClickAnalyticsWrapper extends Component {
+    static propTypes = {
+      hits: PropTypes.array.isRequired,
+      results: PropTypes.object.isRequired,
+    };
+
+    constructor(props) {
+      super(props);
+      this.ref = null;
+    }
+
+    componentDidMount() {
+      const node = this.ref.getDOMNode();
+      node.addEventListener('click', ev => {
+        const event = ev.target.getAttribute('data-analytics-event');
+        const payload = ev.target.getAttribute('data-analytics-payload');
+        if (event && payload) {
+          // event should be oneOf ['clickedObjectIDsAfterSearch', ...]
+          // payload should be valid json
+          let jsonPayload;
+          try {
+            jsonPayload = JSON.parse(payload);
+          } catch (e) {
+            console.error('Invalid JSON event payload');
+            return;
+          }
+
+          // call aa(eventType, eventPayload)
+          console.log(event, jsonPayload);
+        }
+      });
+    }
+
+    render() {
+      const {
+        results: { index, hitsPerPage, page, queryID },
+      } = this.props;
+
+      const hits = this.props.hits.map((hit, position) => ({
+        ...hit,
+        __hitAbsoluteIndex: hitsPerPage * page + position + 1,
+        __index: index,
+        __queryID: queryID,
+        analytics: {
+          clickedObjectIDsAfterSearch: eventName => {
```

Comment:
If I understand your point well, this would require to [pass `results` to the Template component](https://github.com/algolia/instantsearch.js/blob/2a416f8b6e5309b564d3f97824e9e447829fa2a7/src/components/Hits/Hits.js#L24-L34) inside data so that we can [use its data in the helpers](https://github.com/algolia/instantsearch.js/blob/70b3ecd1690f26482166ddc129adf541a06b52f7/src/lib/utils.js#L162-L166), which also means exposing the `results` in every hit. Is this what you had in mind?